### PR TITLE
remove vault encrypter and enable KMS for china

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/giantswarm/issues/7774
On AWS we should only use KMS to don't have snowflakes. No need for multiple encrypters.
* removing vault encrypter
* use  docker instead of rkt
*  use registryDomain from installation config

First, I was thinking we can simply change from `vault` Encrypter to `KMS` encrypter in installation config but there is hardcoded awscli image that is pulled super slow from china so that would not really help. 

Since we have no way how to easily patch already existing VOO tags the simplest solution is to change it only for the `11.0.x` release version and higher. They anyway need to migrate to those clusters so we should not have any migration problems as well.

### CODE REVIEW

**Comment by calvix** [comment]
looks good

From node:
```
 #: sudo cat /opt/bin/decrypt-keys-assets 
#!/bin/bash -e
kms_keys_assets_decrypt() {
AWS_CLI_IMAGE="registry-intl.cn-shanghai.aliyuncs.com/giantswarm/awscli:1.18.3"

while ! docker pull ${AWS_CLI_IMAGE};
do
        echo "Failed to fetch docker image ${AWS_CLI_IMAGE}, retrying in 5 sec."
        sleep 5s
done
echo "Successfully fetched docker image ${AWS_CLI_IMAGE}."


docker run --net=host -v /etc/kubernetes/encryption:/etc/kubernetes/encryption \
        --entrypoint=/bin/sh \
        ${AWS_CLI_IMAGE} \
        -ec \
        'echo decrypting tls assets
    for encKey in $(find /etc/kubernetes/encryption -name "*.enc"); do
      echo decrypting $encKey
      f=$(mktemp $encKey.XXXXXXXX)
      aws \
        --region cn-north-1 kms decrypt \
        --ciphertext-blob fileb://$encKey \
        --output text \
        --query Plaintext \
      | base64 -d > $f
      mv -f $f ${encKey%.enc}
    done;'
}

main() {
  kms_keys_assets_decrypt
  chown -R etcd:etcd /etc/kubernetes/ssl/etcd
}

main
```


```
-- Logs begin at Wed 2020-02-26 12:08:53 UTC, end at Wed 2020-02-26 12:22:22 UTC. --
Feb 26 12:09:52 ip-10-1-4-57.cn-north-1.compute.internal systemd[1]: Starting Decrypt Keys assets...
Feb 26 12:09:53 ip-10-1-4-57.cn-north-1.compute.internal decrypt-keys-assets[2558]: 1.18.3: Pulling from giantswarm/awscli
Feb 26 12:09:53 ip-10-1-4-57.cn-north-1.compute.internal decrypt-keys-assets[2558]: Digest: sha256:b9f87fcde5609a752144e7f9cd08761568f3c0c3c8b3bd165bb536701034b275
Feb 26 12:09:53 ip-10-1-4-57.cn-north-1.compute.internal decrypt-keys-assets[2558]: Status: Image is up to date for registry-intl.cn-shanghai.aliyuncs.com/giantswarm/awscli:1.18.3
Feb 26 12:09:53 ip-10-1-4-57.cn-north-1.compute.internal decrypt-keys-assets[2558]: Successfully fetched docker image registry-intl.cn-shanghai.aliyuncs.com/giantswarm/awscli:1.18.3.
Feb 26 12:09:53 ip-10-1-4-57.cn-north-1.compute.internal decrypt-keys-assets[2558]: decrypting tls assets
Feb 26 12:09:53 ip-10-1-4-57.cn-north-1.compute.internal decrypt-keys-assets[2558]: decrypting /etc/kubernetes/encryption/k8s-encryption-config.yaml.enc
Feb 26 12:09:54 ip-10-1-4-57.cn-north-1.compute.internal systemd[1]: decrypt-keys-assets.service: Succeeded.
Feb 26 12:09:54 ip-10-1-4-57.cn-north-1.compute.internal systemd[1]: Started Decrypt Keys assets.
```

```
-- Logs begin at Wed 2020-02-26 12:08:53 UTC, end at Wed 2020-02-26 12:23:08 UTC. --
Feb 26 12:09:29 localhost systemd[1]: Starting Decrypt TLS certificates...
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 1.18.3: Pulling from giantswarm/awscli
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: e7c96db7181b: Pulling fs layer
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 799a5534f213: Pulling fs layer
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 70a0ae45a681: Pulling fs layer
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 346eb29b0173: Pulling fs layer
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 6af94e8ea8cf: Pulling fs layer
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: e7c96db7181b: Waiting
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 363aadd6e975: Pulling fs layer
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 44704d1aa7b7: Pulling fs layer
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 799a5534f213: Waiting
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 346eb29b0173: Waiting
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 70a0ae45a681: Waiting
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 6af94e8ea8cf: Waiting
Feb 26 12:09:30 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 44704d1aa7b7: Waiting
Feb 26 12:09:33 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: e7c96db7181b: Verifying Checksum
Feb 26 12:09:33 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: e7c96db7181b: Download complete
Feb 26 12:09:33 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: e7c96db7181b: Pull complete
Feb 26 12:09:33 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 799a5534f213: Download complete
Feb 26 12:09:33 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 799a5534f213: Pull complete
Feb 26 12:09:35 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 346eb29b0173: Download complete
Feb 26 12:09:35 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 6af94e8ea8cf: Verifying Checksum
Feb 26 12:09:35 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 6af94e8ea8cf: Download complete
Feb 26 12:09:36 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 363aadd6e975: Verifying Checksum
Feb 26 12:09:36 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 363aadd6e975: Download complete
Feb 26 12:09:36 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 70a0ae45a681: Verifying Checksum
Feb 26 12:09:36 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 70a0ae45a681: Download complete
Feb 26 12:09:38 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 44704d1aa7b7: Verifying Checksum
Feb 26 12:09:38 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 44704d1aa7b7: Download complete
Feb 26 12:09:38 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 70a0ae45a681: Pull complete
Feb 26 12:09:38 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 346eb29b0173: Pull complete
Feb 26 12:09:38 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 6af94e8ea8cf: Pull complete
Feb 26 12:09:38 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 363aadd6e975: Pull complete
Feb 26 12:09:41 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: 44704d1aa7b7: Pull complete
Feb 26 12:09:41 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: Digest: sha256:b9f87fcde5609a752144e7f9cd08761568f3c0c3c8b3bd165bb536701034b275
Feb 26 12:09:41 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: Status: Downloaded newer image for registry-intl.cn-shanghai.aliyuncs.com/giantswarm/awscli:1.18.3
Feb 26 12:09:41 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: Successfully fetched docker image registry-intl.cn-shanghai.aliyuncs.com/giantswarm/awscli:1.18.3.
Feb 26 12:09:47 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting tls assets
Feb 26 12:09:47 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting /etc/kubernetes/ssl/apiserver-crt.pem.enc
Feb 26 12:09:48 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting /etc/kubernetes/ssl/apiserver-ca.pem.enc
Feb 26 12:09:49 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting /etc/kubernetes/ssl/etcd/server-ca.pem.enc
Feb 26 12:09:49 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting /etc/kubernetes/ssl/etcd/client-ca.pem.enc
Feb 26 12:09:50 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting /etc/kubernetes/ssl/etcd/client-key.pem.enc
Feb 26 12:09:50 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting /etc/kubernetes/ssl/etcd/server-key.pem.enc
Feb 26 12:09:50 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting /etc/kubernetes/ssl/etcd/server-crt.pem.enc
Feb 26 12:09:51 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting /etc/kubernetes/ssl/etcd/client-crt.pem.enc
Feb 26 12:09:51 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting /etc/kubernetes/ssl/service-account-key.pem.enc
Feb 26 12:09:52 ip-10-1-4-57.cn-north-1.compute.internal decrypt-tls-assets[1834]: decrypting /etc/kubernetes/ssl/apiserver-key.pem.enc
Feb 26 12:09:52 ip-10-1-4-57.cn-north-1.compute.internal systemd[1]: decrypt-tls-assets.service: Succeeded.
Feb 26 12:09:52 ip-10-1-4-57.cn-north-1.compute.internal systemd[1]: Started Decrypt TLS certificates.

```

**Comment by calvix** [comment]
Works for both China and non-china.

**Comment by lachlanh** [comment]
have had a look through, seems good. we will need to make these changes in ha-masters branches (removing encrypter from tccpn)

File: service/controller/resource/tcnp/create_test.go [hunk 1]
```diff
@@ -31,28 +30,20 @@ var update = flag.Bool("update", false, "update .golden CF template file")
...
-			ctx:              unittest.DefaultContext(),
-			cr:               unittest.DefaultMachineDeployment(),
-			encrypterBackend: encrypter.KMSBackend,
-		},
-		{
-			name:             "case 1: basic test with encrypter backend Vault",
-			ctx:              unittest.DefaultContext(),
-			cr:               unittest.DefaultMachineDeployment(),
-			encrypterBackend: encrypter.VaultBackend,
+			name: "case 0: basic test with encrypter backend KMS",
```
Comment (xh3b4sd) [inline service/controller/resource/tcnp/create_test.go]: That is just a basic test now and has nothing to do with encrypters anymore, as this is also not configurable anymore. 

```suggestion
			name: "case 0: basic test",
```

File: service/controller/resource/tccp/create_test.go [hunk 2]
```diff
@@ -36,76 +35,39 @@ var update = flag.Bool("update", false, "update .golden CF template file")
...
+		errorMatcher   func(error) bool
 	}{
 		{
-			name:             "case 0: basic test without encryption key when encrypter backend kms, route53 enabled",
-			cr:               unittest.DefaultCluster(),
-			ctx:              unittest.DefaultContext(),
-			encrypterBackend: encrypter.KMSBackend,
-			errorMatcher:     nil,
-			route53Enabled:   true,
+			name:           "case 0: basic test without encryption key when encrypter backend kms, route53 enabled",
```
Comment (xh3b4sd) [inline service/controller/resource/tccp/create_test.go]:
```suggestion
			name:           "case 0: basic test without encryption key, route53 enabled",
```

File: service/controller/resource/tccp/create_test.go [hunk 3]
```diff
@@ -36,76 +35,39 @@ var update = flag.Bool("update", false, "update .golden CF template file")
...
+			route53Enabled: true,
 		},
 		{
-			name:             "case 1: basic test with encryption key when encrypter backend kms, route53 enabled",
-			cr:               unittest.DefaultCluster(),
-			ctx:              updateEncryptionKey(unittest.DefaultContext(), "8y5ck"),
-			encrypterBackend: encrypter.KMSBackend,
-			errorMatcher:     nil,
-			route53Enabled:   true,
+			name:           "case 1: basic test with encryption key when encrypter backend kms, route53 enabled",
```
Comment (xh3b4sd) [inline service/controller/resource/tccp/create_test.go]:
```suggestion
			name:           "case 1: basic test with encryption key, route53 enabled",
```

File: service/controller/resource/tccp/create_test.go [hunk 4]
```diff
@@ -36,76 +35,39 @@ var update = flag.Bool("update", false, "update .golden CF template file")
...
+			route53Enabled: true,
 		},
 		{
-			name:             "case 2: basic test without encryption key when encrypter backend vault, route53 enabled",
-			cr:               unittest.DefaultCluster(),
-			ctx:              unittest.DefaultContext(),
-			encrypterBackend: encrypter.VaultBackend,
-			errorMatcher:     nil,
-			route53Enabled:   true,
+			name:           "case 2: basic test without encryption key when encrypter backend kms, route53 disabled",
```
Comment (xh3b4sd) [inline service/controller/resource/tccp/create_test.go]:
```suggestion
			name:           "case 2: basic test without encryption key, route53 disabled",
```

File: service/controller/resource/tccp/create_test.go [hunk 5]
```diff
@@ -36,76 +35,39 @@ var update = flag.Bool("update", false, "update .golden CF template file")
...
-			route53Enabled:   false,
-		},
-		{
-			name:             "case 7: basic test with encryption key when encrypter backend vault, route53 disabled",
-			cr:               unittest.DefaultCluster(),
-			ctx:              updateEncryptionKey(unittest.DefaultContext(), "8y5ck"),
-			encrypterBackend: encrypter.VaultBackend,
-			errorMatcher:     nil,
-			route53Enabled:   false,
+			name:           "case 3: basic test with encryption key when encrypter backend kms, route53 disabled",
```
Comment (xh3b4sd) [inline service/controller/resource/tccp/create_test.go]:
```suggestion
			name:           "case 3: basic test with encryption key, route53 disabled",
```

**Pull request merged**